lib/src/clients.dart

```diff
@@ -856,8 +856,20 @@ Future<http.StreamedResponse> _validateResponse(
       var jsonResponse = await stringStream.transform(json.decoder).first;
       if (jsonResponse is Map && jsonResponse['error'] is Map) {
         final Map error = jsonResponse['error'];
-        final code = error['code'] as int;
+        final codeValue = error['code'];
         final message = error['message'] as String;
+
+        int code;
+        if (codeValue is String) {
+          code = int.tryParse(codeValue) ?? 0;
+          if (code == null) {
```

Comment:
the `?? 0` will ensure `code` is **NEVER** null, right?

Reply:
So for the string value like this "code":"ErrorGettingBrokerSubscription" code 0 will be returned? I know this project is meant to support mainly Google API's so I am not sure if it's ok to ask for more here but ideally the 'code' variable could simply be kept 'dynamic'. Or maybe "ErrorGettingBrokerSubscription" value copied to an additional variable, i.e. textCode or status.

It would be useful for our code to have access to the original textual error code.

Thank you.

Reply:
@sigurdm We can just add an optional field to `ApiRequestError` containing the entire `jsonResponse`. wdyt?

Reply:
Fine with me ... I'll update this patch

Reply:
> the ?? 0 will ensure code is NEVER null, right?

haha - yes, that was not the intention. Thanks for catching that!

Reply:
@mkustermann I added it to the DetailedApiRequestError as an optional named parameter.
@yoxel does this work for you?